File: 66. Plus One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        en = 1
        for i in xrange(len(digits)-1, -1, -1):
            temp = digits[i] + en
            if temp < 10:
                digits[i] = temp
                return digits
            else:
                digits[i] = temp - 10
                en = 1
        return digits if en == 0 else [1] + digits

File: 66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int en = 1;
        for (int i = digits.size()-1; i >= 0; i--)
        {
            int temp = digits[i] + en;
            if (temp < 10)
            {
                digits[i] = temp;
                return digits;
            }
            else
                digits[i] = temp - 10;
        }
        digits.insert(digits.begin(), 1);
        return digits;
    }
};
